core/app/models/spree/stock_item.rb

```diff
@@ -6,12 +6,15 @@ class StockItem < Spree::Base
     belongs_to :variant, class_name: 'Spree::Variant', inverse_of: :stock_items
     has_many :stock_movements, inverse_of: :stock_item
 
-    validates_presence_of :stock_location, :variant
-    validates_uniqueness_of :variant_id, scope: [:stock_location_id, :deleted_at]
+    validates :stock_location, :variant, presence: true
+    validates :variant_id, uniqueness: { scope: [:stock_location_id, :deleted_at] }, allow_blank: true
```

Comment:
Ah, this is valid. allow_blank can be passed directly into the `UniquenessValidator` or used as an option to the `validates` method. The difference being whether or not the validators are invoked, or are expected to handle `:allow_blank` directly (which uniqueness does). I've only ever passed allow_blank into specific validators, I didn't know you could short circuit using it outside of those validation statements. :+1: